OpenSim/Simulation/Model/ContactGeometry.h

```diff
@@ -33,120 +33,84 @@ namespace OpenSim {
-     */
-    OpenSim::PhysicalFrame& updBody();
-    /**
-     * %Set the Body this geometry is attached to.
-     */
-    void setBody(PhysicalFrame& body);
-    /**
-     * Get the name of the Body this geometry is attached to.
-     */
-    const std::string& getBodyName();
-    /**
-     * %Set the name of the Body this geometry is attached to.  This will cause the
-     * Body to be set to NULL, then resolved when extendConnectToModel() is called.
+    /** This constructor connects this ContactGeometry to the provided `frame`,
+     * and uses the default location and orientation (both `Vec3(0)`).
+     *
+     * @param frame        the PhysicalFrame this geometry is attached to;
      */
-    void setBodyName(const std::string& name);
+    explicit ContactGeometry(const PhysicalFrame& frame);
+
     /**
-     * Get the display_preference of this geometry.
+     * @param location     the location of the geometry expressed in `frame`
+     * @param orientation  the orientation of the geometry expressed in `frame`
+     *                     as XYZ body-fixed Euler angles.
+     * @param frame        the PhysicalFrame this geometry is attached to;
+     *                     this constructor connects this ContactGeometry to
+     *                     the provided `frame`
      */
+    ContactGeometry(const SimTK::Vec3& location,
+                    const SimTK::Vec3& orientation,
+                    const PhysicalFrame& frame);
+
+
+    // ACCESSORS
+    /** Get the PhysicalFrame this geometry is attached to. */
+    const PhysicalFrame& getFrame() const;
+    /** %Set the PhysicalFrame this geometry is attached to. */
+    void setFrame(const PhysicalFrame& body);
+
+    /** Get the display_preference of this geometry. */
     const int getDisplayPreference();
-    /**
-     * %Set the display_preference of this geometry.
-     */
+    /** %Set the display_preference of this geometry. */
     void setDisplayPreference(const int dispPref);
-    /**
-     * Create a new SimTK::ContactGeometry based on this object.
-     */
+
+    /** Create a new SimTK::ContactGeometry based on this object. */
     virtual SimTK::ContactGeometry createSimTKContactGeometry() = 0;
-    /**
-     * Get a Transform representing the position and orientation of the geometry
-     * within the Body it is attached to.
-     */
-    SimTK::Transform getTransform();
+
+    /** Get a Transform representing the position and orientation of the
+     * geometry within the base Frame (e.g., a Body or Ground) it is attached
+     * to (*not* the "frame" that the geometry is connected to). If `B` is the
+     * base or Body Frame, `F` is the PhysicalFrame that this geometry is
+     * connected to, and `P` is the frame that is defined (relative to `F`) by
+     * the location and orientation properties of this class, then this returns
+     * `X_BP = X_BF * X_FP`. */
+    SimTK::Transform findTransformInBaseFrame() const;
```

Comment:
I agree with your points (about the scope of the ContactGeometry class). My concern is reducing the chance that users introduce a bug in their code because they have to do some transform multiplication themselves. I'll try to remedy that with good doxygen comments. I will:

1) remove `findTransformInBaseFrame()`.
2) create `findTransformInFrame()`
3) give a doxygen comment to `findTransformInFrame()` that describes how to get "transform in base frame" (this would still involve mentioning base/Ground/Body frames in the header for ContactGeometry).
4) as for the old `getTransform()` method, I assume your preference is to delete it but my preference would be to implement it so that it still has the same functionality but deprecate it. Let me know what you think.

And yeah; we can discuss in person (should be quicker that way); I just wanted to get down these thoughts before I forgot them.